### ISSUE TITLE
Live bindings don't work in some cases

### ISSUE THREAD

**Issue by t532** [rollup/rollup#2754]
- Rollup Version: latest (v1.6.0)
- Operating System (or Browser): macOS Mojave (production), Google Chrome v74.0.3702.0-dev-64bits (REPL)
- Node Version: v11.7.0

### How Do We Reproduce?
- [REPL Link 1](https://rollupjs.org/repl?version=1.6.0&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMiUyRiUyRiUyMG1haW4uanMlNUNuaW1wb3J0JTIwJTdCJTIwY291bnQlMkMlMjBpbmNyZW1lbnQlMjAlN0QlMjBmcm9tJTIwJy4lMkZ3cmFwcGVyJyUzQiU1Q24lNUNuY29uc29sZS5sb2coY291bnQpJTNCJTIwJTJGJTJGJTIwMCU1Q25pbmNyZW1lbnQoKSUzQiU1Q25jb25zb2xlLmxvZyhjb3VudCklM0IlMjAlMkYlMkYlMjAxJTVDbiUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMmluY3JlbWVudGVyLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMiUyRiUyRiUyMGluY3JlbWVudGVyLmpzJTVDbmV4cG9ydCUyMGxldCUyMGNvdW50JTIwJTNEJTIwMCUzQiU1Q24lNUNuZXhwb3J0JTIwZnVuY3Rpb24lMjBpbmNyZW1lbnQoKSUyMCU3QiU1Q24lMjAlMjBjb3VudCUyMCUyQiUzRCUyMDElM0IlNUNuJTdEJTIyJTJDJTIyaXNFbnRyeSUyMiUzQWZhbHNlJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMndyYXBwZXIuanMlMjIlMkMlMjJjb2RlJTIyJTNBJTIyJTJGJTJGJTIwd3JhcHMlMjBpbmNyZW1lbnRlci5qcyU1Q25leHBvcnQlMjAlN0IlMjBjb3VudCUyQyUyMGluY3JlbWVudCUyMCU3RCUyMGZyb20lMjAnLiUyRmluY3JlbWVudGVyJyUzQiUyMiUyQyUyMmlzRW50cnklMjIlM0FmYWxzZSU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJjanMlMjIlMkMlMjJuYW1lJTIyJTNBJTIybXlCdW5kbGUlMjIlMkMlMjJhbWQlMjIlM0ElN0IlMjJpZCUyMiUzQSUyMiUyMiU3RCUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlN0QlMkMlMjJleGFtcGxlJTIyJTNBbnVsbCU3RA==)
- [REPL Link 2](https://rollupjs.org/repl?version=1.6.0&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMiUyRiUyRiUyMG1haW4uanMlNUNuaW1wb3J0JTIwJTdCJTIwY291bnQlMkMlMjBpbmNyZW1lbnQlMjAlN0QlMjBmcm9tJTIwJy4lMkZ3cmFwcGVyJyUzQiU1Q24lNUNuY29uc29sZS5sb2coY291bnQpJTNCJTIwJTJGJTJGJTIwMCU1Q25pbmNyZW1lbnQoKSUzQiU1Q25jb25zb2xlLmxvZyhjb3VudCklM0IlMjAlMkYlMkYlMjAxJTVDbiUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMmluY3JlbWVudGVyLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMiUyRiUyRiUyMGluY3JlbWVudGVyLmpzJTVDbmV4cG9ydCUyMGxldCUyMGNvdW50JTIwJTNEJTIwMCUzQiU1Q24lNUNuZXhwb3J0JTIwZnVuY3Rpb24lMjBpbmNyZW1lbnQoKSUyMCU3QiU1Q24lMjAlMjBjb3VudCUyMCUyQiUzRCUyMDElM0IlNUNuJTdEJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIyd3JhcHBlci5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjIlMkYlMkYlMjB3cmFwcyUyMGluY3JlbWVudGVyLmpzJTVDbmV4cG9ydCUyMCU3QiUyMGNvdW50JTJDJTIwaW5jcmVtZW50JTIwJTdEJTIwZnJvbSUyMCcuJTJGaW5jcmVtZW50ZXInJTNCJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyY2pzJTIyJTJDJTIybmFtZSUyMiUzQSUyMm15QnVuZGxlJTIyJTJDJTIyYW1kJTIyJTNBJTdCJTIyaWQlMjIlM0ElMjIlMjIlN0QlMkMlMjJnbG9iYWxzJTIyJTNBJTdCJTdEJTdEJTJDJTIyZXhhbXBsZSUyMiUzQW51bGwlN0Q=)

### Expected Behavior
According to [the Docs](https://rollupjs.org/guide/en#how-bindings-work), the output code will log
```
0
1
```
to the console.

### Actual Behavior
The codes in the 2 REPLs are identical, and the only difference is that in one REPL `incrementer.js` and `wrapper.js` are treated as *entry module*s and not in the other.

In the first REPL the code compiled and ran as expected:
```js
'use strict';

// incrementer.js
let count = 0;

function increment() {
  count += 1;
}

// wraps incrementer.js

// main.js

console.log(count); // 0
increment();
console.log(count); // 1
```
```
0
1
```

However in the second REPL the output is
```
0
0
```
since `wrapper.js` is compiled to something like this
```js
var incrementer = require('../incrementer.js');

exports.count = incrementer.count;
exports.increment = incrementer.increment;
```
Therefore `import { count } from './wrapper.js'` is no longer a reference bound to that in `increment.js`, but an independent value in `wrapper.js`,  

and `import { increment } from './wrapper.js` still increases the value in `increment.js`, instead of the one in `wrapper.js`.

> This will not happen if you're not assigning a value to an exported variable directly (e.g. assigning to its properties, etc).

> BTW: Babel can compile such code properly, because it defines getters (instead of values) on `exports`:
> ```js
> Object.defineProperty(exports, "count", {
>   enumerable: true,
>   get: function get() {
>     return _incrementer.count;
>   }
> });
> Object.defineProperty(exports, "increment", {
>   enumerable: true,
>   get: function get() {
>     return _incrementer.increment;
>   }
> });
>
> var _incrementer = require("./incrementer");
> ```

> TypeScript has the same problem: [Microsoft/TypeScript#12522](https://github.com/Microsoft/TypeScript/issues/12522)

**Comment by lukastaegert** [comment]
Thanks a lot! I think this is a really old issue from before we had code-splitting which would trigger if you reexport a mutable binding from an external dependency. After some initial thought I think this might be addressed by using a getter on the exports object that always returns the current value when read (and writing is forbidden by the spec anyway). Hope to put some more thought into this soon, especially since this would be a chance to be ahead of TypeScript or even offer a way of fixing TypeScript 😜

**Comment by lukastaegert** [comment]
Fix at #2765
